spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/neo4j/Neo4jProperties.java

```diff
@@ -116,6 +123,13 @@ private void configure(Builder builder) {
 		if (this.username != null && this.password != null) {
 			builder.credentials(this.username, this.password);
 		}
+
+		builder.autoIndex(this.indexes.getAuto().name());
+
+		if (AutoIndexMode.DUMP == this.indexes.getAuto()) {
```

Comment:
I don't know the feature but this looks weird to me. What are the other options of specifying a directory and a filename for the generated index? Why are you checking for "dump" specifically?